libethash-cl/CLMiner.cpp

```diff
@@ -188,9 +188,7 @@ void CLMiner::workLoop()
 			}
 
 			// Read results.
-			// TODO: could use pinned host pointer instead.
-			uint32_t results[c_maxSearchResults + 1];
-			m_queue.enqueueReadBuffer(m_searchBuffer, CL_TRUE, 0, sizeof(results), &results);
+			uint32_t *results = (uint32_t*)m_queue.enqueueMapBuffer(m_searchBuffer, CL_TRUE, CL_MAP_READ, 0, (c_maxSearchResults +1) * sizeof(uint32_t));
```

Comment:
I think you have not changed anything effectively. buffer read is buffer map + buffer unmap. To use pinned memory you have to add `CL_MEM_ALLOC_HOST_PTR` to the buffer itself.